src/networking/systems.rs

```diff
@@ -0,0 +1,240 @@
+        .await
+        .map_err(|e| error!("Connection Error: {:?}", e))?;
+
+    // Send an intial event indicating that this connection opened
+    event_sender
+        .send(ReceiveEvent::Connected(guid))
+        .expect("Failed to send network event");
+
+    // Keep getting events from the connection until it closes
+    while let Some(stream) = bi_streams.next().await {
+        match stream {
+            Err(quinn::ConnectionError::ApplicationClosed { .. }) => break,
+            Err(e) => error!("Connection error: {:?}", e),
+            Ok((send, recv)) => {
+                tokio::spawn(handle_stream(guid, send, recv, event_sender.clone()));
+            }
+        };
+    }
+
+    // Send a final event indicating that this connection closed
+    event_sender
+        .send(ReceiveEvent::Disconnected(guid))
+        .expect("Failed to send network event");
+
+    Ok(())
+}
+
+// Handle all the work of a specific stream
+async fn handle_stream(
+    connection_id: ConnectionId,
+    stream_send: SendStream<TlsSession>,
+    stream_recv: RecvStream<TlsSession>,
+    event_sender: UnboundedSender<ReceiveEvent>,
+) {
+    // Generate a unique ID for this connection
+    let stream_id = StreamId::new();
+    info!(
+        "Stream incoming: conn:{:?} stream:{:?}",
+        connection_id, stream_id
+    );
+
+    // Create an mpsc for sending messages through this stream
+    let (send, recv) = unbounded_channel();
+
+    // Send an intial event indicating that this stream opened, along with the mpsc to send to this stream
+    event_sender
+        .send(ReceiveEvent::OpenedStream {
+            connection_id,
+            stream_id,
+            sender: send,
+        })
+        .expect("Failed to send network event");
+
+    // Spawn a task which pulls packets from this stream and publishes them to the event_sender
+    tokio::spawn(recv_from_stream(
+        connection_id,
+        stream_id,
+        stream_recv,
+        event_sender,
+    ));
+
+    // Spawn a task which pulls messages from the ECS and sends them to this stream
+    tokio::spawn(send_to_stream(stream_send, recv));
+}
+
+// Pull packets from socket and push into an mpsc
+async fn recv_from_stream(
+    connection_id: ConnectionId,
+    stream_id: StreamId,
+    recv: RecvStream<TlsSession>,
+    event_sender: UnboundedSender<ReceiveEvent>,
+) {
+    let mut recv = BoundedPlanetRecvStream::new(recv);
+
+    loop {
+        let pkt = recv.recv_packet().await;
+        let se = match pkt {
+            Ok(pkt) => event_sender.send(ReceiveEvent::ReceivedPacket {
+                connection_id,
+                stream_id,
+                data: pkt,
+            }),
+            Err(err) => event_sender.send(ReceiveEvent::ReceiveError {
+                connection_id,
+                stream_id,
+                err,
+            }),
+        };
+        se.expect("Failed to send event");
+    }
+}
+
+// Pull packets from an mpsc and send them to the given stream
```

Comment:
```suggestion
/// Pull packets from an mpsc and send them to the given stream
```
Might as well turn this into a doc comment since you already did the work to explain it